### PR TITLE
[#4140] improvement(core): Optimize the privileges of securable objects

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Optimize the privileges of securable objects

### Why are the changes needed?

Fix: #4140

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing tests.

### CODE REVIEW

File: api/src/main/java/com/datastrato/gravitino/authorization/Privilege.java [hunk 1]
```diff
@@ -43,72 +43,52 @@ public interface Privilege {
...
-    USE_SCHEMA(0L, 1L << 7),
+    USE_SCHEMA(0L, 1L << 4),
     /** The privilege to create a table. */
-    CREATE_TABLE(0L, 1L << 8),
-    /** The privilege to drop a table. */
-    DROP_TABLE(0L, 1L << 9),
+    CREATE_TABLE(0L, 1L << 5),
     /** The privilege to write a table. */
-    WRITE_TABLE(0L, 1L << 10),
+    MODIFY_TABLE(0L, 1L << 6),
```
[the remark this answers is not in the document]
Comment (jerqi) [reply api/src/main/java/com/datastrato/gravitino/authorization/Privilege.java:55]: MODIFY_TABLE including alter table, insert table, update table, delete table.

**Comment by jerryshao** [comment]
@xunliu would you please help to review this PR, to see if it is OK for you.

**Comment by jerryshao** [comment]
I think you should clearly define the scope and how it take effect for each privilege. Currently, the comment is too simple, even wrong.

File: api/src/main/java/org/apache/gravitino/authorization/Privilege.java [hunk 2]
```diff
@@ -43,66 +43,52 @@ public interface Privilege {
...
+    READ_FILESET(0L, 1L << 10),
     /** The privilege to create a topic. */
-    CREATE_TOPIC(0L, 1L << 16),
-    /** The privilege to drop a topic. */
-    DROP_TOPIC(0L, 1L << 17),
+    CREATE_TOPIC(0L, 1L << 11),
     /** The privilege to write a topic. */
-    WRITE_TOPIC(0L, 1L << 18),
+    PRODUCE_TOPIC(0L, 1L << 12),
     /** The privilege to read a topic. */
```
[the remark this answers is not in the document]
Comment (jerqi) [reply api/src/main/java/org/apache/gravitino/authorization/Privilege.java]: Done.

File: api/src/main/java/org/apache/gravitino/authorization/Privilege.java [hunk 3]
```diff
@@ -43,66 +43,52 @@ public interface Privilege {
...
-    USE_SCHEMA(0L, 1L << 7),
+    USE_SCHEMA(0L, 1L << 4),
     /** The privilege to create a table. */
-    CREATE_TABLE(0L, 1L << 8),
-    /** The privilege to drop a table. */
-    DROP_TABLE(0L, 1L << 9),
+    CREATE_TABLE(0L, 1L << 5),
     /** The privilege to write a table. */
-    WRITE_TABLE(0L, 1L << 10),
+    MODIFY_TABLE(0L, 1L << 6),
```
[the remark this answers is not in the document]
Comment (jerqi) [reply api/src/main/java/org/apache/gravitino/authorization/Privilege.java:55]: Done.

File: api/src/main/java/org/apache/gravitino/authorization/Privilege.java [hunk 4]
```diff
@@ -43,66 +43,52 @@ public interface Privilege {
...
     /** The privilege to read a table. */
-    READ_TABLE(0L, 1L << 11),
+    SELECT_TABLE(0L, 1L << 7),
```
[the remark this answers is not in the document]
Comment (jerqi) [reply api/src/main/java/org/apache/gravitino/authorization/Privilege.java:57]: Done.

File: api/src/main/java/org/apache/gravitino/authorization/Privilege.java [hunk 5]
```diff
@@ -43,66 +43,52 @@ public interface Privilege {
...
     /** The privilege to create a topic. */
-    CREATE_TOPIC(0L, 1L << 16),
-    /** The privilege to drop a topic. */
-    DROP_TOPIC(0L, 1L << 17),
-    /** The privilege to write a topic. */
-    WRITE_TOPIC(0L, 1L << 18),
-    /** The privilege to read a topic. */
-    READ_TOPIC(0L, 1L << 19),
+    CREATE_TOPIC(0L, 1L << 11),
+    /** The privilege to produce a topic. */
```
Comment (jerryshao) [inline api/src/main/java/org/apache/gravitino/authorization/Privilege.java:66]: "to produce to a topic" and "to consume from a topic".

File: api/src/main/java/org/apache/gravitino/authorization/Privilege.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jerqi) [reply api/src/main/java/org/apache/gravitino/authorization/Privilege.java:55]: I corrected it. I means that we use SQL `DELETE` for a table.

File: api/src/main/java/org/apache/gravitino/authorization/Privilege.java (hunk 5 above)
Comment (jerqi) [reply api/src/main/java/org/apache/gravitino/authorization/Privilege.java]: Done.

**Review by jerryshao — APPROVED** [review]
LGTM.

**Comment by jerryshao** [comment]
@xunliu @jerqi I just merged this PR, if you have any more comments or modifications, you can update in the follow-up PRs.

**Pull request merged**